packages/utils/lib/telemetry/metrics.ts

```diff
@@ -135,10 +135,10 @@ export function time<T, E, F extends (...args: E[]) => Promise<T>>(metricName: T
         const start = process.hrtime();
 
         try {
-            const res = func(...args);
-            if (res[Symbol.toStringTag] === 'Promise') {
+            const res = func(...args) as any;
+            if (typeof res === 'object' && res && res[Symbol.toStringTag] === 'Promise') {
                 return res.then(
-                    (v) => {
```

Comment:
Am I the only one or the code after looks worse than the code before

Reply:
ahah, kind of, that's the pain of managing unknows correctly in typescript